### PR TITLE
Waypoint Flight State

### LINKED ISSUE #75: Waypoint State Logic
# Waypoint State Logic

## Problem

Need the code for the Waypoint state of the flight state machine to read from the data structure containing the waypoint GPS data and move the drone to each waypoint in the list. 

## Solution

Read the file containing the GPS points and utilize the move_to() logic to have the drone fly the full flight path. Ensure the drone reaches within 25ft of each point.

## Additional Information

Since we don't have the competition waypoints yet, use a test file with waypoints at the S&T golf course as a stand-in.

### PR DESCRIPTION
**Pull request by DJDMVP**
Flight state for drone moving through each of the waypoints.

### CODE REVIEW

**Comment by fallscameron01** [comment]
Change title to something better and add a description

File: flight/extract_gps.py [hunk 1]
```diff
@@ -27,6 +27,12 @@ class Waypoint(NamedTuple):
     longitude: float
     altitude: float
 
+    def __aiter__(self) -> NamedTuple:
+        return self
```
Comment (cjhr95) [inline flight/extract_gps.py]: Docstring here plz

File: flight/extract_gps.py [hunk 2]
```diff
@@ -27,6 +27,12 @@ class Waypoint(NamedTuple):
...
+
+    async def __anext__(self) -> None:
+        raise StopAsyncIteration
```
Comment (cjhr95) [inline flight/extract_gps.py]: Docstring here again

File: flight/states/waypoints.py [hunk 3]
```diff
@@ -29,4 +31,10 @@ async def run(self, drone: System) -> ODLC:
         ODLC : State
             Re-fly the waypoints if we failed to reach a waypoint boundary, or progress to ODLC flight stage
         """
+        gps_dict: GPSData = extract_gps("flight/data/waypoint_data.json")
+        waypoints: list[Waypoint] = gps_dict["waypoints"]
+
+        for waypoint in waypoints:
+            await move_to(drone, waypoint[0], waypoint[1], waypoint[2], 2 / 3)
```
Comment (cjhr95) [inline flight/states/waypoints.py]: Add a comment here explaining my 2/3 is the float you use for the fast_param

**Review by cjhr95 — CHANGES_REQUESTED** [review]
Documentation

**Review by cjhr95 — APPROVED** [review]
Seems alright I GUESS.

**Review by cjhr95 — APPROVED** [review]
Seems alright

**Pull request merged**